include/oneapi/dpl/internal/binary_search_extension_defs.h

```diff
@@ -25,6 +25,17 @@ namespace oneapi
 {
 namespace dpl
 {
+namespace internal
+{
+
+enum class search_algorithm
```

Comment:
Does it really needed to have only one brick `struct custom_brick` + user visible `enum class search_algorithm` + parametrizations by their enum values? What about three bricks without  `enum class search_algorithm` at all?

Reply:
I'm sorry, but I don't quite follow the question. I have updated the forward declaration of `custom_brick` to now use a forward declared `search_algorithm` enum class. Are you suggesting a different approach?